lib/tasks/seek_upgrades.rake

```diff
@@ -176,6 +178,48 @@ namespace :seek do
     puts "...Updated #{investigations_updated.to_s} investigations"
   end
 
+  task(implement_assay_streams_for_isa_assays: [:environment]) do
+    puts '... Organising isa json compliant assays in assay streams'
+    assay_streams_created = 0
+    disable_authorization_checks do
+      # find assays linked to a study through their sample_types
+      # Should be isa json compliant
+      # Shouldn't already have an assay stream (don't update assays that have been updated already)
+      # Previous ST should be second ST of study
+      first_assays_in_stream = Assay.joins(:sample_type)
+        .select(&:is_isa_json_compliant?)
+        .select { |a| a.assay_stream_id.nil? && (a.previous_linked_sample_type == a.study.sample_types.second) }
```

Comment:
I would try and narrow down this query if you can to avoid needing to load every assay into memory.

I think you most of this filtering at the database level by doing something like:
`Assay.joins(:sample_type, study: :investigation).where(assay_stream_id :nil, investigation: { is_isa_json_compliant: true })`